components/experimental/src/dimension/units/formatter.rs

```diff
@@ -0,0 +1,150 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Experimental.
+
+use fixed_decimal::FixedDecimal;
+use icu_decimal::options::FixedDecimalFormatterOptions;
+use icu_decimal::FixedDecimalFormatter;
+use icu_plurals::PluralRules;
+use icu_provider::DataPayload;
+
+use super::super::provider::units::UnitsDisplayNameV1Marker;
+use super::format::FormattedUnit;
+use super::options::UnitsFormatterOptions;
+use icu_provider::prelude::*;
+
+extern crate alloc;
+
+/// A formatter for monetary values.
+///
+/// [`UnitsFormatter`] supports:
+///   1. Rendering in the locale's units system.
+///   2. Locale-sensitive grouping separator positions.
+///
+/// Read more about the options in the [`super::options`] module.
+pub struct UnitsFormatter {
+    /// Options bag for the units formatter to determine the behavior of the formatter.
+    /// for example: width of the units.
+    options: UnitsFormatterOptions,
+
+    // /// Essential data for the units formatter.
+    // essential: DataPayload<UnitsEssentialsV1Marker>,
+    /// Display name for the units.
+    display_name: DataPayload<UnitsDisplayNameV1Marker>,
+
+    /// A [`FixedDecimalFormatter`] to format the unit value.
+    fixed_decimal_formatter: FixedDecimalFormatter,
+
+    /// A [`PluralRules`] to determine the plural category of the unit.
+    plural_rules: PluralRules,
+}
+
+impl UnitsFormatter {
+    icu_provider::gen_any_buffer_data_constructors!(
+        (locale, unit: &str, options: super::options::UnitsFormatterOptions) -> error: DataError,
+        functions: [
+            try_new: skip,
+            try_new_with_any_provider,
+            try_new_with_buffer_provider,
```

Comment:
Thought: Technically cross-crate constructors should use custom logic, but let's not worry about that at the time being because it's a little annoying

Reply:
sure, I will keep this in mind